CRM/Contact/BAO/Group.php

```diff
@@ -1017,20 +1015,9 @@ public static function getGroupList(&$params) {
         $contactUrl = CRM_Utils_System::url('civicrm/contact/view', "reset=1&cid={$object->created_id}");
         $values[$object->id]['created_by'] = "<a href='{$contactUrl}'>{$object->created_by}</a>";
       }
-    }
 
-    // Get group counts - executes one query for regular groups and another for smart groups
-    foreach ($groupsToCount as $table => $groups) {
-      $where = "g.group_id IN (" . implode(',', $groups) . ")";
-      if ($table == 'civicrm_group_contact') {
-        $where .= " AND g.status = 'Added'";
-      }
-      // Exclude deleted contacts
-      $where .= " and c.id = g.contact_id AND c.is_deleted = 0";
-      $dao = CRM_Core_DAO::executeQuery("SELECT g.group_id, COUNT(*) as `count` FROM $table g, civicrm_contact c WHERE $where GROUP BY g.group_id");
-      while ($dao->fetch()) {
-        $values[$dao->group_id]['count'] = $dao->count;
-      }
+      // get group contact count using Contact.GetCount API
+      $values[$object->id]['count'] = civicrm_api3('Contact', 'getcount', array('group' => $object->id));
```

Comment:
I found that for 'getcount' API action, the ```limit``` is not used [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L4410)